### ISSUE TITLE
Error in mockserver with query param when uses $ref

### ISSUE THREAD

**Issue by jgonzalezcloudappi** [microcks/microcks#439]
When you want to create a mockserver with multiple examples based on a query param. If the query param is a $ ref it doesn't work correctly.

Example good:
  / resources:
```yaml
    get:
      tags: [examples]
      summary: Get resources by query
      description: Get a list of resources by query
      operationId: queryOp
      parameters:
        - name: resourceType
          in: query
          description: Get a list of resources by type
          schema:
            type: string
            enum: [standard, premium]
          examples:
            standard_example:
              value: 'standard'
            premium_example:
              value: 'premium'
      responses:
        '200':
          description: List of resources
          content:
            application / json:
              schema:
                properties:
                  yam:
                    type: string
                  resourceType:
                    type: string
                    enum: [standard, premium]
              examples:
                standard_example:
                  value:
                    - name: 'Resource One'
                      resourceType: 'standard'
                premium_example:
                  value:
                    - name: 'Resource Two'
                      resourceType: 'premium'
                all_example:
                  value:
                    - name: 'Resource One'
                      resourceType: 'standard'
                    - name: 'Resource Two'
                      resourceType: 'premium'
```
Example bad:
  / resources:
```yaml
    get:
      tags: [examples]
      summary: Get resources by query
      description: Get a list of resources by query
      operationId: queryOp
      parameters:
        - $ ref: '# / components / parameters / resourceTypeQueryParam'
      responses:
        '200':
          description: List of resources
          content:
            application / json:
              schema:
                properties:
                  yam:
                    type: string
                  resourceType:
                    type: string
                    enum: [standard, premium]
              examples:
                standard_example:
                  value:
                    - name: 'Resource One'
                      resourceType: 'standard'
                premium_example:
                  value:
                    - name: 'Resource Two'
                      resourceType: 'premium'
                all_example:
                  value:
                    - name: 'Resource One'
                      resourceType: 'standard'
                    - name: 'Resource Two'
                      resourceType: 'premium'
```
Thanks for all.

**Comment by lbroudoux** [comment]
Hi @jgonzalezcloudappi !

What do you mean exactly by "it doesn't work correctly" ? We've got some unit tests covering that case ([here](https://github.com/microcks/microcks/blob/master/webapp/src/test/resources/io/github/microcks/util/openapi/param-refs-openapi.yaml) and [there](https://github.com/microcks/microcks/blob/master/webapp/src/test/java/io/github/microcks/util/openapi/OpenAPIImporterTest.java#L877)) and they seem to be correct.

Could you please provide a full OpenAPI spec sample and detail what is the result you're expecting vs the result you acutally get ? 

Thanks,

**Comment by jgonzalezcloudappi** [comment]
Hi lbroudoux,

I'will provide you a complete example:

`openapi: "3.0.0"
info:
  title: API-Template2
  version: 1.0.0

servers:
  - url: https://dev-api.openfinance.es/api-name/v1
    description: Development server
  - url: https://api.openfinance.es/api-name/v1
    description: Production server
tags:
  - name: examples
    description: 'Examples for Microcks.io'

paths:
  # Example with only one query parameter
  /resources:
    get:
      tags: [examples]
      summary: Get resources by query
      description: Get a list of resources by query
      operationId: queryOp
      parameters:
        - name: resourceType
          in: query
          description: Get a list of resources by type
          schema:
            type: string
            enum:  [standard, premium]
          examples:
            standard_example:
              value: 'standard'
            premium_example:
              value: 'premium'
      responses:
        '200':
          description: List of resources
          content:
            application/json:
              schema:
                properties:
                  name:
                    type: string
                  resourceType:
                    type: string
                    enum: [standard, premium]
              examples:
                standard_example:
                  value:
                    - name: 'Resource One'
                      resourceType: 'standard'
                premium_example:
                  value:
                    - name: 'Resource Two'
                      resourceType: 'premium'
                all_example:
                  value:
                    - name: 'Resource One'
                      resourceType: 'standard'
                    - name: 'Resource Two'
                      resourceType: 'premium'
  # Example with problem in URI_PARAMS
  /accounts:
    get:
      tags: [examples]
      summary: Get resources by multiple query params
      description: Get a list of resources by multiple query params
      operationId: dobleQueryOp
      parameters:
        - $ref: '#/components/parameters/levelQueryParam'
      responses:
        '200':
          description: List of accounts
          content:
            application/json:
              schema:
                properties:
                  accountID:
                    type: string
                  level:
                    type: string
                    enum: [bronze, golden]
              examples:
                bronze_example:
                  value:
                    - accountID: AC001
                      level: bronze
                      region: national
                    - accountID: AC002
                      level: bronze
                      region: international
                golden_example:
                  value:
                    - accountID: AC003
                      level: golden
                      region: national
                    - accountID: AC004
                      level: golden
                      region: international
                all_example:
                  value:
                    - accountID: AC001
                      level: bronze
                      region: national
                    - accountID: AC002
                      level: bronze
                      region: international
                    - accountID: AC003
                      level: golden
                      region: national
                    - accountID: AC004
                      level: golden
                      region: international
                
components:
  parameters:
    levelQueryParam:
      name: level
      in: query
      description: Account level
      schema:
        type: string
        enum: [bronze, golden]
      examples:
        bronze_example:
          value: 'bronze'
        golden_example:
          value: 'golden'
        all_example:
          value: 'all'`

Check in the resource accounts please, uri_params not works correctly

Thanks for all

**Comment by lbroudoux** [comment]
Thanks. Will test this asap.

**Comment by lbroudoux** [comment]
You're right: we've got a bug detecting query parameters using a `$ref`. Fix is on its way ;-) 
Thank you,

**Comment by jgonzalezcloudappi** [comment]
When will it be available, in hotfix or in the new release?

**Comment by lbroudoux** [comment]
It should be available now in latest container image. It will be included into 1.3.0 release I hope on next week.
